### PR TITLE
Do not allow for missing secret key data to cause an NPE

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
Check for null data and do not decrypt it.
Properly synchronize and safe-copy the key array.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java [hunk 1]
```diff
@@ -88,8 +93,8 @@
      *
      * @return byte[] secret data
      */
-    public byte[] getSecretKeyData() {
-        return mSecretKeyData.get();
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:92]: So are we no longer using getters and setters? Do we still need them?

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:92]: we're using them.  Just adding synchronization/safe-copy.

Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java:92]: Ohh okay, so `Arrays.copyOf()` acts as a copy + get.....hadn't used this one before

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java [hunk 2]
```diff
@@ -590,7 +590,13 @@ public SecretKey loadSecretKey(@NonNull final KeyType keyType) throws IOExceptio
...
 
             case ADAL_USER_DEFINED_KEY:
-                return getSecretKey(AuthenticationSettings.INSTANCE.getSecretKeyData());
+                final byte[] secretKeyData = AuthenticationSettings.INSTANCE.getSecretKeyData();
+                if (secretKeyData == null) {
+                    Logger.error("StorageHelper:loadSecretKey", "Went looking for a key that wasn't there.", null);
+                    return null;
+                }
+                SecretKey key = getSecretKey(secretKeyData);
+                return key;
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java:599]:
```suggestion
                return getSecretKey(secretKeyData);
```

**Review by shahzaibj — APPROVED** [review]
Just one suggestion.....LGTM

**Review by andrewchoi5 — APPROVED** [review]

**Pull request merged**